test/core/OptionsPremiumPricer.ts

```diff
@@ -0,0 +1,336 @@
+
+const { provider, getContractFactory } = ethers;
+
+moment.tz.setDefault("UTC");
+
+describe("OptionsPremiumPricer", () => {
+  let mockOracle: Contract;
+  let optionsPremiumPricer: Contract;
+  let wethPriceOracle: Contract;
+  let signer: SignerWithAddress;
+
+  const PERIOD = 43200; // 12 hours
+  const WEEK = 604800; // 7 days
+
+  const ethusdcPool = "0x8ad599c3A0ff1De082011EFDDc58f1908eb6e6D8";
+  const weth = "0xC02aaA39b223FE8D0A0e5C4F27eAD9083C756Cc2";
+  const usdc = "0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48";
+
+  const wethPriceOracleAddress = "0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419";
+  const wethCTokenAddress = "0x4ddc2d193948926d02f9b1fe9e1daa0718270ed5";
+
+  before(async function () {
+    [signer] = await ethers.getSigners();
+    const TestVolOracle = await getContractFactory("TestVolOracle", signer);
+    const OptionsPremiumPricer = await getContractFactory(
+      "OptionsPremiumPricer",
+      signer
+    );
+
+    mockOracle = await TestVolOracle.deploy(ethusdcPool, weth, usdc, PERIOD);
+    optionsPremiumPricer = await OptionsPremiumPricer.deploy(
+      mockOracle.address,
+      wethPriceOracleAddress,
+      wethCTokenAddress
+    );
+    wethPriceOracle = await ethers.getContractAt(
+      "IPriceOracle",
+      await optionsPremiumPricer.priceOracle()
+    );
+  });
+
+  describe.skip("getAssetRiskFreeRate", () => {
+    time.revertToSnapshotAfterEach();
+
+    it("gets reasonable APY (~0%-0.75%)", async function () {
+      const apy = await optionsPremiumPricer.getAssetRiskFreeRate();
+      console.log(apy);
+    });
+  });
+
+  describe("getUnderlyingPrice", () => {
+    time.revertToSnapshotAfterEach();
+
+    it("gets the correct underlying price for asset", async function () {
+      assert.deepEqual(
+        await optionsPremiumPricer.getUnderlyingPrice(),
+        (await wethPriceOracle.latestAnswer()).div(
+          BigNumber.from(10).pow(await wethPriceOracle.decimals())
+        )
+      );
+    });
+  });
+
+  describe("getPremium", () => {
+    time.revertToSnapshotAfterEach();
+
+    beforeEach(async () => {
+      await updateVol();
+    });
+
+    it("reverts on timestamp being in the past", async function () {
+      const expiryTimestamp = (await time.now()).sub(WEEK);
+      await expect(
+        optionsPremiumPricer.getPremium(0, expiryTimestamp, true)
+      ).to.be.revertedWith("Expiry must be in the future!");
+    });
+
+    it("gets the correct premium", async function () {
+      const underlyingStrikeDiff = 200;
+      const strikePrice = BigNumber.from(
+        await optionsPremiumPricer.getUnderlyingPrice()
+      )
+        .add(underlyingStrikeDiff)
+        .mul(BigNumber.from(10).pow(8));
+      const expiryTimestamp = (await time.now()).add(WEEK);
+      const isPut = true;
+
+      const premium = await optionsPremiumPricer.getPremium(
+        strikePrice,
+        expiryTimestamp,
+        isPut
+      );
+
+      assert.isAbove(
```

Comment:
Would be helpful to include `assert.equal` tests so we know what the outputs are